templates/components/workflows/python/deep_research/workflow.py

```diff
@@ -0,0 +1,534 @@
+import logging
+import os
+import uuid
+from typing import List, Literal, Optional
+
+from app.index import IndexConfig, get_index
+from llama_index.core.base.llms.types import (
+    CompletionResponse,
+    CompletionResponseAsyncGen,
+)
+from llama_index.core.indices.base import BaseIndex
+from llama_index.core.memory import ChatMemoryBuffer
+from llama_index.core.memory.simple_composable_memory import SimpleComposableMemory
+from llama_index.core.prompts import PromptTemplate
+from llama_index.core.schema import MetadataMode, Node, NodeWithScore
+from llama_index.core.settings import Settings
+from llama_index.core.types import ChatMessage, MessageRole
+from llama_index.core.workflow import (
+    Context,
+    Event,
+    StartEvent,
+    StopEvent,
+    Workflow,
+    step,
+)
+from llama_index.server.api.models import SourceNodesEvent
+from pydantic import BaseModel, Field
+
+logger = logging.getLogger("uvicorn")
+logger.setLevel(logging.INFO)
+
+
+def create_workflow() -> Workflow:
+    index_config = IndexConfig()
+    index = get_index(index_config)
+    if index is None:
+        raise ValueError(
+            "Index is not found. Try run generation script to create the index first."
+        )
+
+    return DeepResearchWorkflow(
+        index=index,
+        timeout=120.0,
+    )
+
+
+# Workflow events
+class PlanResearchEvent(Event):
+    pass
+
+
+class ResearchEvent(Event):
+    question_id: str
+    question: str
+    context_nodes: List[NodeWithScore]
+
+
+class CollectAnswersEvent(Event):
+    question_id: str
+    question: str
+    answer: str
+
+
+class ReportEvent(Event):
+    pass
+
+
+# Events that are streamed to the frontend and rendered there
+class DeepResearchEventData(BaseModel):
+    event: Literal["retrieve", "analyze", "answer"]
+    state: Literal["pending", "inprogress", "done", "error"]
+    id: Optional[str] = None
+    question: Optional[str] = None
+    answer: Optional[str] = None
+
+
+class DataEvent(Event):
```

Comment:
as llamaindexserver knows this event, probably move it better there (see discusion)